server.go

```diff
@@ -5300,6 +5316,33 @@ func (s *server) SendCustomMessage(peerPub [33]byte, msgType lnwire.MessageType,
 	return peer.SendMessageLazy(true, msg)
 }
 
+// SendOnionMessage sends a custom message to the peer with the specified
```

Comment:
IIUC, you'll do this in a follow up PR, but I think the `actor` package can fit rather nicely here. 

We have two actor types: the processor, and the endpoint. 

The processor handles the crypto logic to figure out who to send the message to next. 

A message endpoint becomes an actor, identifiable by the public key of the node. 

When we read an onion message off the wire, we send it to the processor actor, who the sends it to the relevant processor actor. 

I have a PR that abstracts out the mailbox for actors, so it can be anything that implements the queue abstraction.